cmd/gapir/cc/main.cpp

```diff
@@ -393,12 +396,16 @@ void android_main(struct android_app* app) {
     }
 
     if (serverIsDone) {
-      unlink(socket_file_path.c_str());
-      alive = false;
+      // Start termination of the app
+      ANativeActivity_finish(app->activity);
+      // Note that we need to keep on polling events, eventually APP_CMD_DESTROY
+      // will pop-up after which app->destroyRequested will be true, enabling us
+      // to properly exit the main loop.
     }
   }
   waiting_thread.join();
-  ANativeActivity_finish(app->activity);
+  GAPID_INFO("End of Graphics API Replay");
+  return;
```

Comment:
We probably do want to unlink the file if it is the server at least.

Reply:
It is in the `app->destroyRequested` block. However, I think it would be better to call unlink after joining the waiting thread, i.e. just before exiting.

Reply:
Thanks, moved unliking to be after the waiting_thread.join()